### PR TITLE
Add more info about borrowed from Postgres structures (RelTag and BufferTag)

### PR DESCRIPTION
**Pull request by ololobus**
Follow up to our discussion in chat. Short description and permalinks to the Postgres source code. If it still makes sense

### CODE REVIEW

File: pageserver/src/repository.rs [hunk 1]
```diff
@@ -176,6 +176,23 @@ pub struct RepositoryStats {
     pub num_getpage_requests: Lsn,
 }
 
+//
+// Every data file in Postgres is uniquely identified with 4 numbers:
```
Comment (hlinnaka) [inline pageserver/src/repository.rs:180]: s/identified with/identified by/, I think

File: pageserver/src/repository.rs [hunk 2]
```diff
@@ -176,6 +176,23 @@ pub struct RepositoryStats {
     pub num_getpage_requests: Lsn,
 }
 
+//
```
Comment (hlinnaka) [inline pageserver/src/repository.rs:179]: By Rust convention, use `///` for comments above structs

**Review by hlinnaka — COMMENTED** [review]
Thanks for working on this!

**Review by ericseppanen — COMMENTED** [review]
Thanks for the help!

When you switch this to rustdoc `///` comments, you'll want to use markdown syntax. So `-` for lists, backticks for `code` formatting, etc.

I'm not sure what this `{:syntax}` does. Is this supposed to trigger some particular formatting?

There are lots of good rustdoc examples out there in the wild, including the entire std library ([example](https://doc.rust-lang.org/src/core/option.rs.html#253)). [This](https://doc.rust-lang.org/stable/rust-by-example/meta/doc.html) page is also a nice intro.

To view your documentation in a browser, try running `cargo doc --no-deps --open`.

**Comment by ololobus** [comment]
@hlinnaka, @ericseppanen, thanks for the comments. I pushed refactored version.

@ericseppanen, thank you for sharing the details. I am still reading The Rust Programming Language book, but this topic seems to be not covered there (or I missed this part), so I was wondering what this triple slash comments state for.

_I'm not sure what this {:syntax} does. Is this supposed to trigger some particular formatting?_

No, I just wanted to show somehow that this `relnode` is replaced with value from the struct.

**Comment by ericseppanen** [comment]
> @hlinnaka, @ericseppanen, thanks for the comments. I pushed refactored version.
> 
> @ericseppanen, thank you for sharing the details. I am still reading The Rust Programming Language book, but this topic seems to be not covered there (or I missed this part), so I was wondering what this triple slash comments state for.

That's surprising! And probably an oversight. I may go open a bug for the Rust Book authors pointing this out.

There's also an entire [book about rustdoc](https://doc.rust-lang.org/rustdoc/). Even though `rustdoc` is the name of the tool that does all the work, you probably never need to run `rustdoc` yourself. It's easier to use `cargo doc` and let it do all the work.

There are lots of good documentation examples around: the [standard library documentation](https://doc.rust-lang.org/std/index.html), or any public crate on docs.rs ([example](https://docs.rs/bytes/1.0.1/bytes/)). Everything you see there is automatically generated using rustdoc, and if you click the "src" link you can see what markdown syntax they used to do it.

**Comment by ololobus** [comment]
Oh, my bad, it is there: https://doc.rust-lang.org/book/ch14-02-publishing-to-crates-io.html#making-useful-documentation-comments. I just have not reached that point yet. Anyway, thanks for useful links and comments :)

**Review by ericseppanen — APPROVED** [review]

**Pull request merged**